### DISCUSSION TITLE
[IDEA] Allow for MAUI/XAML items to stay on-screen

### DISCUSSION THREAD

**Discussion by imadofficial** [dotnet/maui#24251] [Ideas]
# Feature description
Have the ability to tap directly into the shell to have 1 (or multiple) elements to stay on-screen regardless if you're going to a different page. That way, it would be easier to display something like a "music buttom bar" or to create your own navbar. Like shown in the video down below:

https://github.com/user-attachments/assets/3810f46e-1cf5-4c01-ac30-fba0bc0570a4

# Maybe some additions?
Maybe add some events too like "OnPageNavigate". And it's very useful to have something like this to increase productivity & so i don't really have to copy & paste frames over & over on every page or copy paste the same code over & over.

**Comment by OvrBtn** [comment]
Doable with some C# and XAML as in the example I've made [https://github.com/OvrBtn/NavigationWrapperExample](https://github.com/OvrBtn/NavigationWrapperExample)


https://github.com/user-attachments/assets/df4723c3-a874-489d-b602-4657fe3e66ee

And you can even play around with making custom navigation animations


https://github.com/user-attachments/assets/ccf5ff6b-c24e-4326-9716-b7799b155d45

**Reply by pictos** [reply]
What you did is really nice, but you aren't reusing the same component, the bottom bar instance from `MainPage` and `SecondPage` aren't the same. I believe what @imadofficial wants is to reuse the same control (instance) on different pages... And it's not possible in Maui right now. If you try to reuse the same control instance on a different `Page`, it will throw an exception saying that control is already inside another `VisualTree`

**Reply by OvrBtn** [reply]
Yes, you are right it's not the same as using the same instance of a control but for use cases like creating custom TabBar it will do the job just fine. Going further with this you can move the static control into separate file to share it between pushed pages on root pages.

Also in terms of reusing same instance - as you said it might throw an exception but what really is the point of using exactly the same object? If one would like to keep the same state of an object across pages it's just the matter of keeping the state data in a static property and then adjusting the control every time it's created. So "music bar button" probably meaining something like Spotify has at the bottom of the app is fully doable in this way.

**Reply by OvrBtn** [reply]
When I think about it I think there is a way to keep exactly the same instance on the screen all the time no matter what happens, but I'd have to test and verify that.

**Reply by OvrBtn** [reply]
In terms of another idea to do this which would keep the same instance on the screen - I don't really have the time to implement it but basically you could use what I've already implemented to achieve transparent page and then write your own class for handling navigation stack operations. And the idea is to always keep one modal page at the top which would contain your static content and when you push/pop a page you'd have to insert it in second place behind the modal page. 
In theory I think it should work just fine. The only thing is that probably it won't be animated but I've already showed how to solved that too.
